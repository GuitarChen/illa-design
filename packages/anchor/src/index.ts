export * from "./anchor"
export * from "./interface"
