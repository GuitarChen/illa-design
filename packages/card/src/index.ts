export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
