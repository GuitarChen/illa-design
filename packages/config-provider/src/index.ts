export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
