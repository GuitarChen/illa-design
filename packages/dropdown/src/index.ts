export * from "./dropdown"
export * from "./interface"
