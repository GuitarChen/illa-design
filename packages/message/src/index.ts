export * from "./style"
export * from "./message"
export * from "./interface"
