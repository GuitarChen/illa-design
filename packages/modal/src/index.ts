export * from "./modal"
export * from "./interface"
export * from "./style"
export * from "./portal"
export * from "./config"
export * from "./confirm"
export * from "./useModal/hookModal"
export * from "./useModal/contextHolder"
export * from "./useModal/index"
