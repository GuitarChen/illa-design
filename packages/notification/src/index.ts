export * from "./notice"
export * from "./interface"
export * from "./notification"
export * from "./style"
