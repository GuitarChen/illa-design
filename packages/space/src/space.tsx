import { Children, forwardRef, Fragment } from "react"
import { SpaceProps } from "./interface"
import { Divider, DividerDirection } from "@illa-design/divider"
import { applyContainer, applyDividerSize } from "./style"

export const Space = forwardRef<HTMLDivElement, SpaceProps>((props, ref) => {
  const {
    size = "small",
    align = "center",
    direction = "horizontal",
    divider,
    wrap,
    ...otherProps
  } = props

  let childrenArray = Children.toArray(props.children)

  return (
    <div
      css={applyContainer(direction, align, wrap ?? false)}
      ref={ref}
      {...otherProps}
    >
      {childrenArray.map((child, index) => {
        return (
          <Fragment key={index}>
            {index != 0 && divider ? (
              <Divider
                css={applyDividerSize(size, direction, wrap ?? false, false)}
                direction={
                  direction == "horizontal"
                    ? ("vertical" as DividerDirection)
                    : ("horizontal" as DividerDirection)
                }
              />
            ) : null}
            <div
              css={applyDividerSize(
                size,
                direction,
                wrap ?? false,
                index == childrenArray.length - 1,
              )}
            >
              {child}
            </div>
          </Fragment>
        )
      })}
    </div>
  )
})

Space.displayName = "Space"
