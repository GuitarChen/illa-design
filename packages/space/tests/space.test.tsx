import { render, screen } from "@testing-library/react"
import { Space, SpaceAlign } from "../src"
import "@testing-library/jest-dom"

test("Space renders with single size", () => {
  render(
    <div>
      <Space size="mini">
        <span data-testid="test-space-mini">A</span>
        <span data-testid="test-space-mini-last">B</span>
      </Space>
      <Space size="small">
        <span data-testid="test-space-small">A</span>
        <span>B</span>
      </Space>
      <Space size="medium">
        <span data-testid="test-space-medium">A</span>
        <span>B</span>
      </Space>
      <Space size="large">
        <span data-testid="test-space-large">A</span>
        <span>B</span>
      </Space>
      <Space size="30px">
        <span data-testid="test-space-custom">A</span>
        <span>B</span>
      </Space>
      <Space size={["30px"]}>
        <span data-testid="test-space-single-element-array">A</span>
        <span data-testid="test-space-single-element-array-last">B</span>
      </Space>
    </div>,
  )
  expect(screen.getByTestId("test-space-mini").parentNode).toHaveStyle({
    "margin-right": "4px",
  })
  expect(screen.getByTestId("test-space-mini-last").parentNode).not.toHaveStyle(
    {
      "margin-right": "4px",
    },
  )
  expect(screen.getByTestId("test-space-small").parentNode).toHaveStyle({
    "margin-right": "8px",
  })
  expect(screen.getByTestId("test-space-medium").parentNode).toHaveStyle({
    "margin-right": "16px",
  })
  expect(screen.getByTestId("test-space-large").parentNode).toHaveStyle({
    "margin-right": "24px",
  })
  expect(screen.getByTestId("test-space-custom").parentNode).toHaveStyle({
    "margin-right": "30px",
  })
  expect(
    screen.getByTestId("test-space-single-element-array").parentNode,
  ).toHaveStyle({
    "margin-right": "30px",
  })
  expect(
    screen.getByTestId("test-space-single-element-array-last").parentNode,
  ).not.toHaveStyle({
    "margin-right": "30px",
  })
})

test("Space renders with size array but without wrap", () => {
  render(
    <Space size={["small", "medium"]}>
      <span data-testid="test-space-with-size-array">A</span>
      <span data-testid="test-space-with-size-array-last">B</span>
    </Space>,
  )
  expect(
    screen.getByTestId("test-space-with-size-array").parentNode,
  ).toHaveStyle({
    "margin-right": "8px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array").parentNode,
  ).not.toHaveStyle({
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last").parentNode,
  ).not.toHaveStyle({
    "margin-bottom": "16px",
    "margin-right": "8px",
  })
})

test("Space renders with size array and vertical but without wrap", () => {
  render(
    <div>
      <Space size={["small", "medium"]} direction="vertical">
        <span data-testid="test-space-with-size-array">A</span>
        <span data-testid="test-space-with-size-array-last">B</span>
      </Space>
    </div>,
  )
  expect(
    screen.getByTestId("test-space-with-size-array").parentNode,
  ).toHaveStyle({
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array").parentNode,
  ).not.toHaveStyle({
    "margin-right": "8px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last").parentNode,
  ).not.toHaveStyle({
    "margin-bottom": "16px",
    "margin-right": "8px",
  })
})

test("Space renders with size array and wrap", () => {
  render(
    <div>
      <Space size={["small", "medium"]} wrap>
        <span data-testid="test-space-with-size-array">A</span>
        <span data-testid="test-space-with-size-array-last">B</span>
      </Space>
      <Space size={["small", "medium"]} wrap direction="vertical">
        <span data-testid="test-space-with-size-array-vertical">A</span>
        <span data-testid="test-space-with-size-array-last-vertical">B</span>
      </Space>
    </div>,
  )
  expect(
    screen.getByTestId("test-space-with-size-array").parentNode,
  ).toHaveStyle({
    "margin-right": "8px",
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last").parentNode,
  ).toHaveStyle({
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last").parentNode,
  ).not.toHaveStyle({
    "margin-right": "8px",
  })

  expect(
    screen.getByTestId("test-space-with-size-array-vertical").parentNode,
  ).toHaveStyle({
    "margin-right": "8px",
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last-vertical").parentNode,
  ).not.toHaveStyle({
    "margin-bottom": "16px",
  })
  expect(
    screen.getByTestId("test-space-with-size-array-last-vertical").parentNode,
  ).toHaveStyle({
    "margin-right": "8px",
  })
})

test("Space renders with empty size array", () => {
  render(
    <div>
      <Space size={[]}>
        <span data-testid="test-space-with-empty-size-array">A</span>
        <span>B</span>
      </Space>
    </div>,
  )
  expect(
    screen.getByTestId("test-space-with-empty-size-array").parentNode,
  ).not.toHaveStyle({
    "margin-right": "8px",
    "margin-bottom": "8px",
  })
})

test("Space renders with different direction", () => {
  render(
    <div>
      <Space direction="horizontal">
        <span data-testid="test-space-horizontal">A</span>
        <span>B</span>
      </Space>
      <Space direction="vertical">
        <span data-testid="test-space-vertical">A</span>
        <span>B</span>
      </Space>
    </div>,
  )
  expect(screen.getByTestId("test-space-horizontal").parentNode).toHaveStyle(
    "margin-right:8px",
  )
  expect(screen.getByTestId("test-space-vertical").parentNode).toHaveStyle(
    "margin-bottom:8px",
  )
})

test("Space renders with different align", () => {
  const alignList: SpaceAlign[] = ["start", "center", "end", "baseline"]
  for (let align of alignList) {
    const { asFragment } = render(
      <Space align={align}>
        <span>A</span>
        <span>B</span>
      </Space>,
    )
    expect(asFragment()).toMatchSnapshot()
  }
})

test("Space renders with wrap", () => {
  const { asFragment } = render(
    <Space wrap={true}>
      <span>A</span>
      <span>B</span>
      <span>C</span>
      <span>D</span>
      <span>E</span>
      <span>F</span>
    </Space>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Space renders with divider", () => {
  const { asFragment } = render(
    <Space divider={true}>
      <span>A</span>
      <span>B</span>
    </Space>,
  )
  expect(asFragment()).toMatchSnapshot()
})
