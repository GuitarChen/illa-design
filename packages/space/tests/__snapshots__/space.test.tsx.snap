// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space renders with different align 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 0px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with different align 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 0px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with different align 3`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 0px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with different align 4`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 0px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with divider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 0px;
}

.emotion-2 {
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #e5e6eb;
}

.emotion-2 span {
  padding: 0 16px;
}

.emotion-4 {
  display: inline-flex;
  vertical-align: middle;
  border-width: 0 0 0 1px;
  border-image: initial;
  border-style: solid;
  height: 1em;
}

.emotion-5 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 0px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-1 emotion-4"
      />
    </div>
    <div
      class="emotion-5"
    >
      <span>
        B
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Space renders with wrap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
  margin-bottom: 8px;
}

.emotion-6 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 0px;
  margin-bottom: 8px;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span>
        A
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <span>
        B
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <span>
        C
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <span>
        D
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <span>
        E
      </span>
    </div>
    <div
      class="emotion-6"
    >
      <span>
        F
      </span>
    </div>
  </div>
</DocumentFragment>
`;
